test/TestApp30.Tests/BasicTestWithCorrelation.cs

```diff
@@ -0,0 +1,318 @@
+﻿using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace TestApp30.Tests
+{
+    public class BasicTestWithCorrelation : IClassFixture<CustomWebApplicationFactory<TestApp30.Startup>>
+    {
+        private readonly CustomWebApplicationFactory<TestApp30.Startup> _factory;
+        protected readonly ITestOutputHelper output;
+
+
+        public BasicTestWithCorrelation(CustomWebApplicationFactory<TestApp30.Startup> factory, ITestOutputHelper output)
+        {
+            this.output = output;
+            _factory = factory;
+            _factory.sentItems.Clear();
+        }
+
+        [Fact]
+        public async Task RequestSuccessWithTraceParent()
+        {            
+            // Arrange
+            var client = _factory.CreateClient();
+            var url = "Home/Index";
+
+            // Act
+            Dictionary<string, string> requestHeaders = new Dictionary<string, string>()
+                {
+                    { "traceparent", "00-4e3083444c10254ba40513c7316332eb-e2a5f830c0ee2c46-00"}
+                };
+            var request = CreateRequestMessage(requestHeaders);
+            request.RequestUri = new Uri(client.BaseAddress + url);
+            var response = await client.SendAsync(request);
+
+            // Assert
+            response.EnsureSuccessStatusCode();
+
+            this.output.WriteLine(response.Content.ReadAsStringAsync().Result);
+
+            Task.Delay(1000).Wait();
```

Comment:
if you have to wait, can you spinlock until sentItems got a new item and add a comment describing why? My first intention if I saw this code would be to remove wait and it would be good to have some explanation that it is needed

Reply:
I added comment explaining the need for Wait here.
Don't know if doing anything more complex than simple wait is worth the effort now. 
Open to suggestions.